src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -984,12 +985,13 @@ def __call__(self, poller):
 
         try:
             result = poller.result()
-        except ClientException as client_exception:
+        except (ClientException, HttpResponseError) as client_exception:
             from azure.cli.core.commands.arm import handle_long_running_operation_exception
             self.progress_bar.stop()
-            if getattr(client_exception, 'status_code', None) == 404 and 'delete' in self.cli_ctx.data['command']:
-                logger.debug('Service returned 404 on the long-running delete operation. CLI treats it as delete '
-                             'successfully but service should fix this behavior.')
+            if getattr(client_exception, 'status_code', None) == 404 and \
+               ('delete' in self.cli_ctx.data['command'] or 'purge' in self.cli_ctx.data['command']):
+                logger.debug('Service returned 404 on the long-running delete or purge operation. CLI treats it as '
+                             'delete or purge successfully but service should fix this behavior.')
                 return None
             handle_long_running_operation_exception(client_exception)
```

Comment:
Please double check whether `handle_long_running_operation_exception` can handle track 2 `HttpREsponseError` as well.

Reply:
Yes, it could. 
Before this change, the `HttpResponseError` is raised to an outer exception handler.
After this change, it just wraps the exception and throw a CLIError, which i think doesn't matter.

Reply:
We may add anthor except block to keep the exactly same behavior, but i think it's not really necessary.